### ISSUE TITLE
Clicking a link to a header when window is narrow makes the header hidden under the top menu bar.

### ISSUE THREAD

**Issue by KristofferC** [JuliaDocs/Documenter.jl#1960]
Try click https://pkgdocs.julialang.org/v1/environments/#**4.**-Working-with-Environments while having a narrow window. The result should be similar to:

![gnome-shell-screenshot-fkc0r5](https://user-images.githubusercontent.com/1282691/194018192-6e216328-a0da-4ac0-a7da-505490d2110f.png)

The header is hidden by the top bar and it is a bit confusing since you expected to see the header you got linked to. Scrolling up makes the header be revealed.

![gnome-shell-screenshot-28jmcg](https://user-images.githubusercontent.com/1282691/194018322-68ac308f-c2a6-4b8e-86a6-9e6b9581391a.png)

It would be good if you could see the header when clicking a link even if the window is narrow.

**Comment by Hetarth02** [comment]
I cannot reproduce this, do we still have this issue?

**Comment by mortenpi** [comment]
I can repro this. E.g. if you navigate to https://documenter.juliadocs.org/stable/#Package-Features

**Comment by Hetarth02** [comment]
Ok, but it isn't fixed right? It moves when scrolling(even at top/bottom of page).

**Comment by mortenpi** [comment]
> Ok, but that header isn't fixed right? It moves when scrolling(even at top/bottom of page).

Do you mean the top bar? The UX problem here is that the heading you'd expect to see is hidden when you navigate to the page. Ideally it should be visible. I guess there are two solutions here:

1. Don't jump quite to the heading, but slightly higher, so that it wouldn't be hidden by the top bar.
2. Just hide the top bar when you load the page.

I actually feel partial to (2) -- seems simpler. But it would be interesting to know how other sites that use headroom handle this.

**Comment by Sisira07** [comment]
Hi!
I was able to reproduce the issue and as per my understanding, a potential fix would be to add a **scroll-margin-top** to the headings, in [_main.scss](https://github.com/JuliaDocs/Documenter.jl/blob/233a7822d670b063e894f62aea00729bcb3a28dd/assets/html/scss/documenter/layout/_main.scss).

The results when you click on a heading appear like:

Before:

<img width="961" height="816" alt="Image" src="https://github.com/user-attachments/assets/788cce93-77bf-4a91-afbf-9151a86c92b3" />

After:

<img width="1027" height="920" alt="Image" src="https://github.com/user-attachments/assets/0f4eeeda-d22a-471e-b140-9a72055d03dc" />


As suggested, this would scroll a little higher than before, so the heading would still be visible.

**Comment by fingolfin** [comment]
I can't tell from your screenshot whether it solves the issue with <https://documenter.juliadocs.org/stable/#Package-Features>, though?

**Comment by Sisira07** [comment]
Just to clarify my understanding of the issue: when clicking a link to a subheading, the page navigates to that section but the heading itself is briefly hidden by the navbar. The heading is visible again after the navbar disappears (which takes less than a second).  

If this is the issue being referred to, then navigating to a position slightly above the subheading will resolve it.
With code, setting the scroll-margin's top to say 4rem (which is the navbar's height) would make sure that the subheading is visible at all times (not even briefly covered by the navbar). 

I experimented this using the browser's inspect tool, however if it's correct i can make changes to the [_main.scss](https://github.com/JuliaDocs/Documenter.jl/blob/233a7822d670b063e894f62aea00729bcb3a28dd/assets/html/scss/documenter/layout/_main.scss) file.

**Comment by fingolfin** [comment]
The issue appears for me if I make the window with this issue very narrow, then click on https://documenter.juliadocs.org/stable/#Package-Features

**Comment by Sisira07** [comment]
Yes, i was referring to the same! So currently the heading isn't visible, I will position the heading by calculating the height of the navbar, and then the heading will appear after that height.

**Comment by Hetarth02** [comment]
@Sisira07 A small screen recording of your changes with before and after should clear up the confusion.

**Comment by Sisira07** [comment]
Yes, sorry about that! Hope this clears it: 

https://github.com/user-attachments/assets/f88a8dd4-7f1f-4868-a754-94c2fc42cd37

As mentioned before, I am using the inspect tool for demonstration. We can see that the heading is no longer hidden after checking the added property (scroll-margin-top).
